atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/impl/expectations/CellCountValidatorTest.java

```diff
@@ -0,0 +1,162 @@
+                    CellCountValidator.validateCacheAndGetNonEmptyValuesCount(0, Map.of());
+
+                    CellCountValidator.validateFetchedLessOrEqualToExpected(1, cellMapOf("foo", "bar"));
+                    CellCountValidator.validateCacheAndGetNonEmptyValuesCount(1, cacheCellMapOf("foo", "bar"));
+                })
+                .doesNotThrowAnyException();
+    }
+
+    @Test
+    public void testThrowsIfFetchedMoreCellsThanExpected() {
+        Map<Cell, byte[]> presentValues = cellMapOf("foo", "bar", "baz", "qux");
+        assertThatLoggableExceptionThrownBy(
+                        () -> CellCountValidator.validateFetchedLessOrEqualToExpected(1, presentValues))
+                .isInstanceOf(MoreCellsPresentThanExpectedException.class)
+                .hasExactlyArgs(
+                        SafeArg.of("expectedNumberOfCells", 1L),
+                        SafeArg.of("numberOfCellsRetrieved", 2),
+                        UnsafeArg.of("retrievedCells", presentValues));
+
+        Map<Cell, CacheValue> cachedValues = toCached(presentValues);
+        assertThatLoggableExceptionThrownBy(
+                        () -> CellCountValidator.validateCacheAndGetNonEmptyValuesCount(1, cachedValues))
+                .isInstanceOf(MoreCellsPresentThanExpectedException.class)
+                .hasExactlyArgs(
+                        SafeArg.of("expectedNumberOfCells", 1L),
+                        SafeArg.of("numberOfCellsRetrieved", 2),
+                        UnsafeArg.of("retrievedCells", presentValues));
+    }
+
+    @Test
+    public void testThrowsIfExpectingNegativeValue() {
+        Map<Cell, byte[]> presentValues = cellMapOf("foo", "bar", "baz", "qux");
+        assertThatLoggableExceptionThrownBy(
+                        () -> CellCountValidator.validateFetchedLessOrEqualToExpected(-1, presentValues))
+                .isInstanceOf(MoreCellsPresentThanExpectedException.class)
+                .hasExactlyArgs(
+                        SafeArg.of("expectedNumberOfCells", -1L),
+                        SafeArg.of("numberOfCellsRetrieved", 2),
+                        UnsafeArg.of("retrievedCells", presentValues));
+
+        Map<Cell, CacheValue> cachedValues = toCached(presentValues);
+        assertThatLoggableExceptionThrownBy(
+                        () -> CellCountValidator.validateCacheAndGetNonEmptyValuesCount(-1, cachedValues))
+                .isInstanceOf(MoreCellsPresentThanExpectedException.class)
+                .hasExactlyArgs(
+                        SafeArg.of("expectedNumberOfCells", -1L),
+                        SafeArg.of("numberOfCellsRetrieved", 2),
+                        UnsafeArg.of("retrievedCells", presentValues));
+    }
+
+    @Test
+    public void testEmptyCacheValuesDoesNotCountAgainstPresentCells() {
+        Map<Cell, CacheValue> cachedValues = Map.of(
+                Cell.create("foo".getBytes(StandardCharsets.UTF_8), "someColumn".getBytes(StandardCharsets.UTF_8)),
```

Comment:
Nit: I would just use `PtBytes::toBytes` and just make some of these `static final`